tests/integration/test_fim/test_files/test_ambiguous_confs/test_whodata_prevails_over_realtime.py

```diff
@@ -59,47 +59,43 @@
 import os
 
 import pytest
-from wazuh_testing import global_parameters
-from wazuh_testing.fim import (LOG_FILE_PATH, generate_params, callback_detect_event,
-                               REGULAR, create_file, delete_file)
+from wazuh_testing import global_parameters, LOG_FILE_PATH, REGULAR
 from wazuh_testing.tools import PREFIX
-from wazuh_testing.tools.configuration import load_wazuh_configurations
+from wazuh_testing.tools.configuration import get_test_cases_data, load_configuration_template
+from wazuh_testing.tools.file import create_file, delete_file
 from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.modules.fim.event_monitor import callback_detect_event
+from wazuh_testing.modules.fim import FIM_DEFAULT_LOCAL_INTERNAL_OPTIONS as local_internal_options
+
 
 # Marks
 pytestmark = pytest.mark.tier(level=2)
 
-# Variables and configuration
-wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
-
-test_directories = [os.path.join(PREFIX, 'testdir1'),
-                    os.path.join(PREFIX, 'testdir2')
-                    ]
-dir1, dir2 = test_directories
-
-test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
-configurations_path = os.path.join(test_data_path, 'wazuh_conf_whodata_prevails_over_realtime.yaml')
+# Variables
+test_directories = [os.path.join(PREFIX, 'testdir1')]
 
-conf_params = {'TEST_DIR1': dir1, 'TEST_DIR2': dir2, 'MODULE_NAME': __name__}
-p, m = generate_params(extra_params=conf_params, modes=['whodata'])
+# Configuration paths
+TEST_DATA_PATH = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'data')
+CONFIGURATIONS_PATH = os.path.join(TEST_DATA_PATH, 'configuration_templates')
+TEST_CASES_PATH = os.path.join(TEST_DATA_PATH, 'test_cases')
 
-configurations = load_wazuh_configurations(configurations_path, __name__, params=p, metadata=m)
+# Configuration and cases data
+test_cases_path = os.path.join(TEST_CASES_PATH, 'cases_whodata_prevails_over_realtime.yaml')
+configurations_path = os.path.join(CONFIGURATIONS_PATH, 'configuration_whodata_prevails_over_realtime.yaml')
 
-
-# Fixture
-@pytest.fixture(scope='module', params=configurations)
-def get_configuration(request):
-    """Get configurations from the module."""
-    return request.param
+# Test configurations
+configuration_parameters, configuration_metadata, test_case_ids = get_test_cases_data(test_cases_path)
+for count, value in enumerate(configuration_parameters):
+    configuration_parameters[count]['TEST_DIR'] = test_directories[0]
+configurations = load_configuration_template(configurations_path, configuration_parameters, configuration_metadata)
 
 
 # Test
-@pytest.mark.parametrize('directory', [
-    dir1,
-    dir2,
-])
-def test_whodata_prevails_over_realtime(directory, get_configuration, put_env_variables, configure_environment,
-                                        restart_syscheckd, wait_for_fim_start):
+@pytest.mark.parametrize('test_folders', [test_directories], scope="module", ids='')
```

Comment:
We need to pass it because the fixture `create_monitored_folders_module` uses it as a parameter. The fixture has a `function` scoped version as well, for when different cases create different folders. We need to pass it and in this case tell it the scope is module because the default scope would be function.